### PR TITLE
GIX-1220: SNS Proposal payload card

### PR DESCRIPTION
**Pull request by lmuntaner**
# Motivation

User can see the payload of an SNS Proposal.

# Changes

* Implement SnsProposalPayloadSection.svelte using ProposalSummary.svelte.

# Tests

* New page object SnsProposalPayloadSectionPo
* Test SnsProposalPayloadSection component with new PO.

### CODE REVIEW

**Comment by lmuntaner** [comment]
The payload comes in markdown text. Therefore, I used the same component we use for the summary, which is also in markdown.

This way we have consistent styles.

PENDING: Discuss with NNS team about the value in the response. It's not user friendly.

New UI:

Screenshot 2023-03-29 at 08 56 53
Screenshot 2023-03-29 at 09 00 21
Screenshot 2023-03-29 at 09 00 35

**Comment by lmuntaner** [comment]
@dskloetd @mstrasinskis please review

File: frontend/src/tests/page-objects/SnsProposalPayloadSection.page-object.ts [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+
+  getCardTitle(): Promise<string> {
+    return this.root.querySelector("h2").getText();
+  }
+
+  getPayloadText(): Promise<string> {
+    return this.root.byTestId("proposal-summary-component").getText();
+  }
+
+  hasContent(): Promise<boolean> {
```
Comment (dskloetd) [inline frontend/src/tests/page-objects/SnsProposalPayloadSection.page-object.ts:25]: BasePageObject has isPresent so you can call isPresent directly on the po and this method is not necessary.

File: frontend/src/tests/page-objects/SnsProposalPayloadSection.page-object.ts [hunk 2]
```diff
@@ -0,0 +1,28 @@
+import { BasePageObject } from "$tests/page-objects/base.page-object";
+import type { PageObjectElement } from "$tests/types/page-object.types";
+
+export class SnsProposalPayloadSectionPo extends BasePageObject {
+  static readonly tid = "sns-proposal-payload-section-component";
```
[the remark this answers is not in the document]
Comment (lmuntaner) [reply frontend/src/tests/page-objects/SnsProposalPayloadSection.page-object.ts:5]: sure

File: frontend/src/tests/lib/components/sns-proposals/SnsProposalPayloadSection.spec.ts [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+    it("should render title", async () => {
+      const po = await renderComponent(props);
+
+      expect(await po.getCardTitle()).toBe(en.proposal_detail.payload);
+    });
+
+    it("should contain summary", async () => {
+      const po = await renderComponent(props);
+
+      expect(await po.getPayloadText()).toContain(payload);
```
[the remark this answers is not in the document]
Comment (lmuntaner) [reply frontend/src/tests/lib/components/sns-proposals/SnsProposalPayloadSection.spec.ts:47]: Some spacing

**Comment by lmuntaner** [comment]
@dskloetd ready for another review

File: frontend/src/tests/lib/components/sns-proposals/SnsProposalPayloadSection.spec.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (dskloetd) [reply frontend/src/tests/lib/components/sns-proposals/SnsProposalPayloadSection.spec.ts:47]: Would `expect(await po.getPayloadText().trim()).toEqual(payload);` work?

[the remark this answers is not in the document]
Comment (lmuntaner) [reply frontend/src/tests/lib/components/sns-proposals/SnsProposalPayloadSection.spec.ts:47]: Yes, thanks! Done!

**Comment by lmuntaner** [comment]
@dskloetd ready for another review.

Couldn't you have approved already with the last comment?

**Pull request merged**